src/signalo/core/tests.py

```diff
@@ -142,34 +144,49 @@ def setUpTestData(cls):
 
         cls.demo_viewer = User.objects.get(username="demo_viewer")
         cls.admin = User.objects.get(username="admin")
+        cls.collection_url = collections_url + "/signalo_core.pole"
+        cls.items_url = cls.collection_url + "/items"
 
     def tearDown(self):
         self.client.force_authenticate(user=None)
 
-    def test_auth_as_demo_viewer(self):
+    def test_get_as_viewer(self):
         collections_from_anonymous = self.client.get(
-            "/oapif/collections", format="json"
+            collections_url, format="json"
         ).json()
         self.client.force_authenticate(user=self.demo_viewer)
-        collection_response = self.client.get("/oapif/collections", format="json")
-        oapif_response_code = self.client.get("/oapif/").status_code
+        collection_response = self.client.get(collections_url, format="json")
 
-        self.assertEqual(oapif_response_code, 200)
         self.assertEqual(collection_response.status_code, 200)
         self.assertEqual(
             len(collection_response.json()), len(collections_from_anonymous)
         )
 
-    def test_auth_as_admin(self):
-        collections_from_anonymous = self.client.get(
-            "/oapif/collections", format="json"
-        ).json()
+    def test_post_as_admin(self):
         self.client.force_authenticate(user=self.admin)
-        collection_response = self.client.get("/oapif/collections", format="json")
-        oapif_response_code = self.client.get("/oapif/").status_code
+        data = {"geom": "Point(1300000 600000)", "name": "test123"}
+        post_to_items = self.client.post(self.items_url, data, format="json")
+        self.assertIn(post_to_items.status_code, (200, 201))
 
-        self.assertEqual(oapif_response_code, 200)
-        self.assertEqual(collection_response.status_code, 200)
-        self.assertEqual(
-            len(collection_response.json()), len(collections_from_anonymous)
-        )
+    def test_anonymous_items_options(self):
+        # Anonymous user
+        expected = {"GET", "OPTIONS", "HEAD"}
+        response = self.client.options(self.items_url)
+
+        allowed_headers = set(s.strip() for s in response.headers["Allow"].split(","))
+        allowed_body = set(response.json()["actions"].keys())
+
+        self.assertEqual(allowed_body, expected)
+        self.assertEqual(allowed_headers, allowed_body)
+
+    def test_admin_items_options(self):
```

Comment:
@3nids This test passes. This means that QGIS will get correct responses to its __authenticated__ OPTIONS requests, doesn't it?

Reply:
hopefully :)